README.md

```diff
@@ -1,4 +1,253 @@
 # CSV Permissions Module for Django
 
-<Insert Readme Here>
+CSV-based permissions for Django.
 
+Read [Motivation / Rationale](doc/background.md) for why this project exists.
+
+
+## System Requirements
+
+* Tested with django 2.2 and 3.2
+  * Pull requests accepted for other versions, but at minimum we test against current LTS versions
+* Python >=3.6 (no python 3.5 support)
+
+## CSV Permissions
+
+The `csv_permissions` model works as follows:
+
+* Every user is expected to have a `user_type` attribute (the equivalent of `django.contrib.auth`'s `Group`)
+
+* A CSV file that defines a permission matrix is used to define what permissions each `user_type` has
+
+### Installation
+
+In your django settings:
+
+* Add `csv_permissions` to `INSTALLED_APPS` (TODO: necessary?)
+
+* Add `csv_permissions.permissions.CSVPermissionsBackend` to `AUTHENTICATION_BACKENDS` 
+  
+* Add a `CSV_PERMISSIONS_PATHS` which is an array/tuple of `str`/`pathlib.Path`
+  pointing to the CSV files you want to use to define your permissions
+
+#### Autoreload
+
+`csv_permissions` caches the data read from the CSV permissions file on server start.
+During development this means you have to manually restart the dev server when you make changes.
+You can hook into django's autoreloader to automatically reload when the CSV file is changed:
+
+In one of your [django app configs](https://docs.djangoproject.com/en/dev/ref/applications/#for-application-authors):
+
+```python
+def add_csv_permissions_watcher(sender, **kwargs):
+    """In dev we want to reload if the csv permission file changes"""
+    sender.extra_files.add(settings.CSV_PERMISSIONS_PATH)
+
+class MySiteAppConfig(AppConfig):
+    name = "my_site"
+    verbose_name = "My Site"
+
+    def ready(self):
+        if settings.DEBUG:
+            from django.utils.autoreload import autoreload_started
+
+            autoreload_started.connect(add_csv_permissions_watcher)
+```
+
+If you're using [`runserver_plus`](https://django-extensions.readthedocs.io/en/latest/runserver_plus.html)
+from `django-extensions` you can add your CSV files to
+[`settings.RUNSERVER_PLUS_EXTRA_FILES`](https://django-extensions.readthedocs.io/en/latest/runserver_plus.html#configuration).   
+
+  
+### The CSV File
```

Comment:
explain how multiple CSV files work, the load order (does not) matter etc

Reply:
added comment